packages/components/app/styles/components/form/common.scss

```diff
@@ -13,7 +13,10 @@
 .hds-form-field--vertical-layout {
   display: grid;
   gap: 4px;
-  width: max-content;
+
+  .hds-form-field__label {
+    width: fit-content; // needed to avoid extra invisible space since the <label> is interactive
```

Comment:
Good catch! I guess it's worth testing under different types of containers after all 😬